include/ArchiveEntry.h

```diff
@@ -0,0 +1,142 @@
+    /// of the most common/basic information that is common between data in most or all of these places. Not
+    /// every archive will support all of the information covered here, but they should contain most of it.
+    ///////////////////////////////////////
+    struct MEZZ_LIB ArchiveEntry
+    {
+        ///////////////////////////////////////////////////////////////////////////////
+        // Construction and Destruction
+
+        /// @brief Class constructor.
+        ArchiveEntry() = default;
+        /// @brief Copy constructor.
+        /// @param Other The other entry to be copied.
+        ArchiveEntry(const ArchiveEntry& Other) = default;
+        /// @brief Move constructor.
+        /// @param Other The other entry to be moved.
+        ArchiveEntry(ArchiveEntry&& Other) = default;
+
+        ///////////////////////////////////////////////////////////////////////////////
+        // Operators
+
+        /// @brief Copy assignment operator.
+        /// @param Other The other entry to be copied.
+        /// @return Returns a reference to this.
+        ArchiveEntry& operator=(const ArchiveEntry& Other) = default;
+        /// @brief Move assignment operator.
+        /// @param Other The other entry to be moved.
+        /// @return Returns a reference to this.
+        ArchiveEntry& operator=(ArchiveEntry&& Other) = default;
+
+        ///////////////////////////////////////////////////////////////////////////////
+        // Type Information
+
+        /// @brief Gets the type of archive this entry belongs to.
+        /// @remarks Generally archives will each have their own Entry type, so this enum can be used to determine
+        /// what this entry can be safely casted to in order to get more detailed metadata about the file.
+        ArchiveType Archive = ArchiveType::Invalid;
+        /// @brief Gets the type of entry this is in the Archive.
+        /// @remarks See the EntryType enum for more detailed description of entry types.
+        EntryType Entry = EntryType::Unknown;
+        /// @brief Gets the method used to compress the file referenced by the entry.
+        CompressionMethod Compression = CompressionMethod::Unknown;
+        /// @brief The algorithm used to encrypt the file referenced by the entry.
+        EncryptionMethod Encryption = EncryptionMethod::Unknown;
+
+        ///////////////////////////////////////////////////////////////////////////////
+        // Basic Metadata
+
+        /// @brief The full path to and name of the file.
+        String Name;
+        /// @brief A short description of the file.
+        String Comment;
+        /// @brief The size of the file.
+        /// @remarks Usually in compressed Archives, this will be the uncompressed size of the file.
+        UInt64 Size = 0;
+        /// @brief The compressed size of the file (if applicable).
+        UInt64 CompressedSize = 0;
+        /// @brief The time the file was created.
+        UInt64 CreateTime = 0;
```

Comment:
Should these timepoints or durations or some other time specific type?

Reply:
These are likely to be values pulled directly from the OS file API.  So there isn't really a dedicated type.

Reply:
Gross. 

So there is no guaranteed clean conversion from what the OS/API gives us to std::timepoint? If not then this is as good as can be done.